File: app/src/main/java/com/example/oderfoodapp/ViewHolder/MenuPopularViewHolder.java
package com.example.oderfoodapp.ViewHolder;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.oderfoodapp.Interface.ItemClickListener;
import com.example.oderfoodapp.R;

public class MenuPopularViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    public TextView txt_category_name;
    public ImageView popular_imageView;

    private ItemClickListener itemClickListener;

    public MenuPopularViewHolder(@NonNull View itemView) {

        super(itemView);
        txt_category_name = itemView.findViewById(R.id.txt_category_name);
        popular_imageView = itemView.findViewById(R.id.category_image);
        itemView.setOnClickListener(this);
    }

    public void setItemClickListener(ItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    @Override
    public void onClick(View view) {
        itemClickListener.onClick(view,getAdapterPosition(),false   );

    }
}

File: app/src/main/java/com/example/oderfoodapp/view/view_holder/MenuPopularViewHolder.kt
package com.example.oderfoodapp.view.view_holder

import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.oderfoodapp.databinding.LayoutPopularCategoriesItemBinding
import com.example.oderfoodapp.model.Category

class MenuPopularViewHolder(val binding: LayoutPopularCategoriesItemBinding) :
    RecyclerView.ViewHolder(binding.root) {


    fun bind(model: Category) {
        binding.txtCategoryName.text = model.name
        Glide.with(binding.root.context).load(model.image)
            .into(binding.categoryImage)
    }

    companion object {
        fun create(binding: LayoutPopularCategoriesItemBinding): MenuPopularViewHolder {
            return MenuPopularViewHolder(binding)
        }
    }
}